parts/k8s/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -225,6 +225,21 @@ apt_get_dist_upgrade() {
   echo Executed apt-get dist-upgrade $i times
   wait_for_apt_locks
 }
+apt_fix_keys() {
+  retries=10
+  output=/tmp/apt-fix-keys.out
+  for i in $(seq 1 $retries); do
+    wait_for_apt_locks
+    ! (apt-get update | tee $output | grep NO_PUBKEY) && \
+    cat $output && break || \
+    apt-key adv --keyserver keyserver.ubuntu.com --recv-keys $(apt-get update | grep NO_PUBKEY -m 1 | awk -F "NO_PUBKEY" '{print $2}')
```

Comment:
@palma21 Can we add `keyserver.ubuntu.com` to the "no egress" whitelist? This URL is used to remotely retrieve expired/missing keys in the apt repo configuration.

Reply:
if we have a cluster with no egress lock down, I assume it doesn't need apt operation, why we need this whitelist?

Reply:
My understanding is there are apt sources in the whitelist to allow no egress clusters to update themselves (kernel, azsecpack, etc).

If not, then AKS Engine is definitely delivering an apt configuration that doesn't make sense in the no egress scenario.